backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -54,18 +54,29 @@ impl JsonParser {
     /// # Return
     /// A boolean if the image has an valid license or not.
     /// If the image has no license or no license history, the image isn't restricted by any license and true will be returned.
-    #[must_use]
-    pub fn check_license(root: &JsonRootLicense) -> bool {
+    /// # Errors
+    /// If an image is invalid [`ImageHosterError::InvalidLicense`] with the invalid license will be returned.
+    pub fn check_license(root: JsonRootLicense) -> Result<(), ImageHosterError> {
         let license = root
             .license_history
-            .iter()
+            .into_iter()
             .max_by_key(|l| l.date_change)
-            .map(|entry| &entry.new_license);
-
-        if let Some(license) = license {
-            return VALID_LICENSES.contains(&license.as_str());
+            .map(|entry| entry.new_license)
+            .ok_or_else(|| {
+                ImageHosterError::InvalidLicense(
+                    String::from("No license could be detected"),
+                    format!("{VALID_LICENSES:?}"),
+                )
+            })?;
+        let str_license = license.as_str();
+        if VALID_LICENSES.contains(&str_license) {
+            Ok(())
+        } else {
+            Err(ImageHosterError::InvalidLicense(
+                license,
+                format!("{VALID_LICENSES:?}"),
+            ))
```

Comment:
Is there a way to pass these arrays without debug print? Or is the only aesthetic solution to implement display?

Reply:
an array of strings can be `.join(delimiter)`ed to a single string